src/minters/signature/ZoraSignatureMinterStrategy.sol

```diff
@@ -0,0 +1,249 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {Enjoy} from "_imagine/mint/Enjoy.sol";
+import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import {IMinter1155} from "../../interfaces/IMinter1155.sol";
+import {ICreatorCommands} from "../../interfaces/ICreatorCommands.sol";
+import {SaleStrategy} from "../SaleStrategy.sol";
+import {ICreatorCommands} from "../../interfaces/ICreatorCommands.sol";
+import {SaleCommandHelper} from "../utils/SaleCommandHelper.sol";
+import {LimitedMintPerAddress} from "../utils/LimitedMintPerAddress.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {IReadableAuthRegistry} from "../../interfaces/IAuthRegistry.sol";
+
+/*
+
+
+             ░░░░░░░░░░░░░░              
+        ░░▒▒░░░░░░░░░░░░░░░░░░░░        
+      ░░▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░      
+    ░░▒▒▒▒░░░░░░░░░░░░░░    ░░░░░░░░    
+   ░▓▓▒▒▒▒░░░░░░░░░░░░        ░░░░░░░    
+  ░▓▓▓▒▒▒▒░░░░░░░░░░░░        ░░░░░░░░  
+  ░▓▓▓▒▒▒▒░░░░░░░░░░░░░░    ░░░░░░░░░░  
+  ░▓▓▓▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░░░░░  
+  ░▓▓▓▓▓▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░░░░  
+   ░▓▓▓▓▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░  
+    ░░▓▓▓▓▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░    
+    ░░▓▓▓▓▓▓▒▒▒▒▒▒▒▒░░░░░░░░░▒▒▒▒▒░░    
+      ░░▓▓▓▓▓▓▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒░░      
+          ░░▓▓▓▓▓▓▓▓▓▓▓▓▒▒░░░          
+
+               OURS TRULY,
+
+
+    github.com/ourzora/zora-1155-contracts
+
+ */
+
+/// @title ZoraSignatureMinterStrategy
+/// @notice Mints tokens based on signature created by an authorized signer
+/// @author @oveddan
+contract ZoraSignatureMinterStrategy is Enjoy, SaleStrategy, LimitedMintPerAddress, EIP712 {
+    using SaleCommandHelper for ICreatorCommands.CommandSet;
+
+    /// @notice General signatue sale settings
+    struct SalesConfig {
+        /// @notice Registry that decides which account is authorized to sign signatures
+        /// to mint for the sale
+        IReadableAuthRegistry authorizedSignatureCreators;
+    }
+
+    // target -> settings
+    mapping(address => SalesConfig) signatureSaleSettings;
```

Comment:
What's the gas savings of hardcoding a default authorized registry when none is explicitly set?

Reply:
Another approach that could be useful is if the address is not a smart contract just use that address as the signer address so individual wallets can use this config without deploying the registry.